packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -0,0 +1,210 @@
+      fireEvent.click(row2);
+    });
+
+    it('should contains the row id', () => {
+      expect(eventArgs!.params.id).to.equal(2);
+    });
+    it('should contains the element', () => {
+      expect(eventArgs!.params.element).to.equal(row2);
+    });
+    it('should contains the row model', () => {
+      expect(eventArgs!.params.row).to.equal(baselineProps.rows[1]);
+    });
+    it('should contains the row index', () => {
+      expect(eventArgs!.params.rowIndex).to.equal(1);
+    });
+    it('should contains the columns', () => {
+      expect(eventArgs!.params.columns).to.equal(apiRef!.current.getAllColumns());
+    });
+    it('should contains the api', () => {
+      expect(eventArgs!.params.api).to.equal(apiRef.current);
+    });
+  });
+
+  describe('CellsParams', () => {
+    let eventArgs: { params: GridCellParams; event: React.MouseEvent } | null = null;
+    let cell22;
+    beforeEach(() => {
+      const handleClick = (params, event) => {
+        eventArgs = { params, event };
+      };
+      render(<TestEvents onCellClick={handleClick} />);
+
+      cell22 = getCell(1, 1);
+      fireEvent.click(cell22);
+    });
+
+    it('should contains the cell id', () => {
+      expect(eventArgs!.params.id).to.equal(2);
+    });
+    it('should contains the cell value', () => {
+      expect(eventArgs!.params.value).to.equal('Jack');
+    });
+
+    it('should consider value getter', () => {
+      const cellFirstAge = getCell(1, 3);
+      fireEvent.click(cellFirstAge);
+
+      expect(eventArgs!.params.value).to.equal('Jack_11');
+    });
+    it('should consider value formatter', () => {
+      const cellFirstAge = getCell(1, 3);
+      fireEvent.click(cellFirstAge);
+
+      expect(eventArgs!.params.formattedValue).to.equal('Jack_11 yrs');
+    });
+    it('should contains the cell formattedValue', () => {
+      expect(eventArgs!.params.formattedValue).to.equal('Jack');
+    });
+    it('should contains the cell isEditable', () => {
+      expect(eventArgs!.params.isEditable).to.equal(true);
+    });
+    it('should contains the element', () => {
+      expect(eventArgs!.params.element).to.equal(cell22);
+    });
+    it('should contains the cell model', () => {
+      expect(eventArgs!.params.row).to.equal(baselineProps.rows[1]);
+    });
+    it('should contains the cell index', () => {
+      expect(eventArgs!.params.rowIndex).to.equal(1);
+    });
+    it('should contains the column', () => {
+      expect(eventArgs!.params.colDef).to.equal(apiRef!.current.getColumnFromField('first'));
+    });
+    it('should contains the column index', () => {
+      expect(eventArgs!.params.colIndex).to.equal(1);
+    });
+    it('should contains the api', () => {
+      expect(eventArgs!.params.api).to.equal(apiRef.current);
+    });
+  });
+
+  describe('onCellClick', () => {
+    it('should bubble to the row', () => {
+      let events = '';
+      const addEventName = (name: string) => () => {
+        events += `${name}, `;
+      };
```

Comment:
It might be "cleaner" to push into an array or to us a spy and compare with a deep equal. But no strong preferences.